### DISCUSSION TITLE
Cost accounting so linkdrop-proxy can be used by anyone

### DISCUSSION THREAD

**Discussion by mattlockyer** [near-apps/linkdrop-proxy#9] [Ideas]
User's creating linkdrop contract will want to reclaim the unused NEAR from the access key allowance.

Proposal is to check the used gas pre/post account being created (assume linkdrop proxy already has self callback to confirm account was created and delete access key)

The sum of the deleted access key's original allowance - 
used gas from `create_account_and_claim` (proxy) - 
attached gas to call `create_account_and_claim` in main linkdrop contract - 
used gas from linkdrop self callback =
total unused allowance amount

Let the creator of the linkdrop withdraw this amount or use this amount to create subsequent linkdrops.

**Comment by willemneal** [comment]
Some terms: LC -> link creator, C -> Claimer, LCB - linkdrop callback contract

So I see this working in two ways:

- Maintain another mapping of `AccoundId -> Balance`, which increases the storage cost for the LC, but then allows the LC to withdraw with a separate method (?)
- LCB tracks deposits and then handles the transfer of deposit on callback (if deposits are even needed). This simplifies the linkdrop's interface since now it doesn't even need to manage the storage of the key or balance, and just has to add/remove the access key. 


Perhaps we can still do both?  That way users like an NFT contract which don't even need the deposit (just the minimum to create the account) can have a cheaper path through the linkdrop contract and others can still use it to get the refund?
